### ISSUE TITLE
Asking for Propagator function in the newest version of Deepwave

### ISSUE THREAD

**Issue by dhhien1503** [ar4/deepwave#56]
Hi Alan,

Thank you for making a great seismic modeling and inversion package. I have followed the code GAN for FWI which he used your wave propagator code for generate the data as given in the following function

def generate_data(truth, source, xs, xr, dx, dt, device="cpu"):
    prop = dw.scalar.Propagator({'vp': truth.to(device)}, dx)
    data = prop(source, xs, xr, dt)
    data = data_normalization(data).permute(1, 0, 2).unsqueeze(0)
    return data.detach()

I think that the deepwave.scalar.Propagator is in the deepwave version (0.6.6) as you used in the example here 
https://colab.research.google.com/drive/1PMO1rFAaibRjwjhBuyH3dLQ1sW5wfec_#scrollTo=zfjDVs6Q17k2
With the newest version that I installed using Pip the function dw.scalar.Propagator does not exit.

Can you please help to share with me the function to generate seismic data similarly in the version 0.6.6 in the newest version that I can simulate the wavefield

Thank and regards,
Hien

**Comment by ar4** [comment]
Hi Hien,

I am delighted to hear that you are interested in Deepwave. Thank you
for writing to me with your question.

You are right that the Deepwave interfaces changed after v0.0.9. It
was unfortunate, but necessary so that it could provide some new
features. The code

prop = deepwave.scalar.Propagator({'vp': model_true.to(device)}, dx)
receiver_amplitudes_true = prop(source_amplitudes_true.to(device),
x_s.to(device),
x_r.to(device), dt).cpu()

from v0.0.6 would now be written

receiver_amplitudes_true = deepwave.scalar(model_true.to(device), dx, dt,
source_amplitudes=source_amplitudes_true.to(device),
source_locations=x_s.to(device),
receiver_locations=x_r.to(device))[-1].cpu()

Note that the propagator now returns a list of tensors, and the
receiver data is the final element of the list (which is why there is
a "[-1]" at the end of the command).

If you have not seen it already, the Deepwave documentation contains
quite a few examples and an explanation of the inputs and outputs:
https://ausargeo.com/deepwave .

I hope that this answers your question, but please let me know if it
is still unclear or if you have other questions.

-Alan

**Comment by dhhien1503** [comment]
Hi Alan,

Thank you very much for your swift reply and clear explaination. I managed to run the code using your supports.

Kind regards,
Hien

**Comment by ar4** [comment]
Hi Hien,

That is wonderful news. Please let me know if you encounter any other issues or have other problems.

-Alan
